src/include/parsing/parsing.h

```diff
@@ -0,0 +1,109 @@
+#ifndef JUCC_PARSING_PARSING_H
+#define JUCC_PARSING_PARSING_H
```

Comment:
parser/parser.h
Why call a class "Parsing" ?

Reply:
Bcoz it performs parsing.

Reply:
include in .cpp file then .h doesn't require this.

Reply:
class should be called Parser